src/mpcCoreKit.ts

```diff
@@ -934,10 +970,182 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
+      remoteFactorPub,
+      metadataShare,
+      remoteClientToken,
+      tssShareIndex,
+      signatures: this.state.signatures,
+    };
+
+    const sharestore = ShareStore.fromJSON(JSON.parse(metadataShare));
+    await this.tkey.inputShareStoreSafe(sharestore);
+    await this.tKey.reconstructKey();
+    const tssPubKey = this.tKey.getTSSPub().toSEC1(this.tkey.tssCurve, false);
+    // setup Tkey
+    // const tssPubKey = Point.fromTkeyPoint(this.tKey.getTSSPub()).toBufferSEC1(false);
+    this.updateState({ tssShareIndex, tssPubKey, remoteClient });
+    // // Finalize setup.
+    // setup provider
+    await this.createSessionRemoteClient();
+  }
+
+  public async remoteSignSecp256k1(msgData: Buffer, hashed: boolean = false): Promise<{ v: number; r: Buffer; s: Buffer }> {
+    if (!hashed) {
+      msgData = keccak256(msgData);
+    }
+
+    if (!this.state.remoteClient.remoteClientUrl) throw new Error("remoteClientUrl not present");
+
+    // PreSetup
+    const { torusNodeTSSEndpoints } = fetchLocalConfig(this.options.web3AuthNetwork, this.keyType);
+
+    const tssCommits = this.tKey.getTSSCommits();
+
+    const tssNonce = this.getTssNonce() || 0;
+
+    const vid = `${this.verifier}${DELIMITERS.Delimiter1}${this.verifierId}`;
+    const sessionId = `${vid}${DELIMITERS.Delimiter2}default${DELIMITERS.Delimiter3}${tssNonce}${DELIMITERS.Delimiter4}`;
+
+    const parties = 4;
+    const clientIndex = parties - 1;
+
+    const { nodeIndexes } = await (this.tKey.serviceProvider as TSSTorusServiceProvider).getTSSPubKey(
+      this.tKey.tssTag,
+      this.tKey.metadata.tssNonces[this.tKey.tssTag]
+    );
+
+    if (parties - 1 > nodeIndexes.length) {
+      throw new Error(`Not enough nodes to perform TSS - parties :${parties}, nodeIndexes:${nodeIndexes.length}`);
+    }
+    const { endpoints, tssWSEndpoints, partyIndexes } = generateTSSEndpoints(torusNodeTSSEndpoints, parties, clientIndex, nodeIndexes);
+
+    const accountNonce = this.tkey.computeAccountNonce(this.state.accountIndex);
+    const factor = Point.fromSEC1(secp256k1, this.state.remoteClient.remoteFactorPub);
+    const factorEnc = this.tKey.getFactorEncs(factor);
+
+    const data: { remoteSignParams: RemoteDklsSignParams; msgHash: string } = {
+      remoteSignParams: {
+        factorEnc,
+        sessionId,
+        tssNonce,
+        accountNonce: accountNonce.toString("hex"),
+        nodeIndexes: nodeIndexes.slice(0, parties - 1),
+        tssCommits: tssCommits.map((commit) => pointToHex(commit)),
+        signatures: this.signatures,
+        serverEndpoints: { endpoints, tssWSEndpoints, partyIndexes },
+        tssPubKeyHex: this.state.tssPubKey.toString("hex"),
+        curve: this.tkey.tssKeyType,
+      },
+      msgHash: msgData.toString("hex"),
+    };
+
+    const result = await post<{ data?: Record<string, string> }>(`${this.state.remoteClient.remoteClientUrl}/api/v3/mpc/sign/dkls`, data, {
```

Comment:
this logic should be moved to a function in plugin so that we can be decoupled from api endpoints here, i think another way to do it is to do it directly inside existing sign function since most of the code is same and call the remoteSign function same as we are doing for copyShare logic.